### PR TITLE
CPU optimization in muonisolation::TrackSelector: postpone eta/phi computation until needed

### PR DESCRIPTION
**Pull request by slava77**
based on https://cms-reco-profiling.web.cern.ch/cms-reco-profiling/cgi-bin/igprof-navigator/releases/CMSSW_12_0_0_pre3/23434.21/step3/cpu_endjob/1249

a small downside is that the debug/trace variant will be a bit slower.

### CODE REVIEW

**Comment by slava77** [comment]
enable profiling

**Comment by slava77** [comment]
@cmsbuild please test

File: RecoMuon/MuonIsolation/plugins/TrackSelector.cc [hunk 1]
```diff
@@ -50,10 +28,12 @@ TrackSelector::result_type TrackSelector::operator()(const TrackSelector::input_
       continue;
     if (!thePars.rRange.inside(tD0Cor))
       continue;
-    if (thePars.dir.deltaR(reco::isodeposit::Direction(tEta, tPhi)) > thePars.drMax)
-      continue;
     if (tChi2Ndof > thePars.chi2NdofMax)
       continue;
+    float tEta = tk.eta();
+    float tPhi = tk.phi();
+    if (thePars.dir.deltaR(reco::isodeposit::Direction(tEta, tPhi)) > thePars.drMax)
```
Comment (perrotta) [inline RecoMuon/MuonIsolation/plugins/TrackSelector.cc:35]: Using `deltaR2` could optimize a bit further

**Comment by slava77** [comment]
abort test

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
+reconstruction

for https://github.com/cms-sw/cmssw/pull/34281	00ef9d7 
- code changes are technical 
- jenkins tests pass and comparisons with the baseline show no (relevant) differences
- the profiling tests show a speedup in PU200 phase2 workflow by about 0.15%
    - before  https://cms-reco-profiling.web.cern.ch/cms-reco-profiling/cgi-bin/igprof-navigator/releases/CMSSW_12_0_0_pre3/23434.21/step3/cpu_endjob/1249
    - after https://cmssdt.cern.ch/SDT/cgi-bin/igprof-navigator/CMSSW_12_0_X_2021-06-29-1100/slc7_amd64_gcc900/profiling/23434.21/PR-851e44/16328/igprofCPU_step3/2559
    - taking `muonisolation::CaloExtractorByAssociator::deposits` as a reference for normalization, the call to `muonisolation::TrackExtractor::deposit` is now about 4.5 times faster. More speedup should be possible, but it will require some more significant code refactoring and the total 0.04% is not motivating enough atm.

**Comment by qliphy** [comment]
+1

**Pull request merged**